### ISSUE TITLE
Using npm to install yarn

### ISSUE THREAD

**Issue by MalcolmDwyer** [eirslett/frontend-maven-plugin#504]
**TLDR**: How do I 'npm install yarn', then run 'yarn install' from that yarn?

(Similar to https://github.com/eirslett/frontend-maven-plugin/issues/503)

### Motivation
I'm trying to set up a yarn build on an isolated corporate system. We have an internal npm repo, and an internal copy of node_dist.  We **don't** have any internal yarn dist though.  By default the 'install-node-and-yarn' execution is getting yarn downloads from https://github.com/yarnpkg/yarn/releases/download/...
However, while there are gzip files reachable there, it's not readable as a directory and therefore, I can't reasonably ask my configuration support people to clone it internally.

I tried pointing the yarnDownloadRoot at our local npm repo.  I can find a path where yarn-0.16.0.tgz exists, but the yarnDownloadRoot argument makes incompatible assumptions about how the version number fits into that path.

### Requesting either of the following:
1. A way to install yarn given an npm repo path, instead of a dist path
2. A way to run yarn from node_modules.  In this scenario (similar to what issue 503 is asking), I would use `<goal>install-node-and-npm</goal>`, then stick in an extra step to run `npm install yarn`, then I would need a way in `<goal>yarn install</goal>` to tell it to use the `node_modules/yarn/bin/yarn` instead of `node/yarn/bin/yarn`.  If there's a `configuration` variable to do this, I couldn't find it.

Thanks

**Comment by eirslett** [comment]
The answer is in #490, I posted an example there.
